src/battle_ai_util.c

```diff
@@ -4887,52 +4893,373 @@ bool32 IsMoxieTypeAbility(u32 ability)
     }
 }
 
-// Should the AI use a spread move to deliberately activate its partner's ability?
-bool32 ShouldTriggerAbility(u32 battler, u32 ability)
+// TODO: work out when to attack into the player's contextually 'beneficial' ability
+bool32 ShouldTriggerAbility(u32 battlerAtk, u32 battlerDef, u32 ability)
 {
-    switch (ability)
+    if (IsTargetingPartner(battlerAtk, battlerDef))
     {
-        case ABILITY_LIGHTNING_ROD:
-        case ABILITY_STORM_DRAIN:
-            if (B_REDIRECT_ABILITY_IMMUNITY < GEN_5)
+        switch (ability)
+        {
+            case ABILITY_LIGHTNING_ROD:
```

Comment:
```suggestion
        switch (ability)
        {
        case ABILITY_LIGHTNING_ROD:
```

we align the case alongside the switch